src/flop.sol

```diff
@@ -135,6 +138,11 @@ contract Flopper is LibNote {
 
         vat.move(msg.sender, bids[id].guy, bid);
 
+        // on first dent, clear Ash
+        if (bids[id].tic == 0) {
+            VowLike(bids[id].guy).kiss(bid);
```

Comment:
I would actually check `Ash` as you had previously suggested. It is true that in normal operations shouldn't happen to not have `Ash` when doing the first `bid`. But wondering if a possible future `Vow` replacement would bring issues here.
If we do the full replacement with `flop` then we should be ok, but if we want to maintain the `flop` for any reason, there we would have an issue.
I'm not really sure about this, but wondering if the extra check might be beneficial just in case. It's true that will add some nasty extra gas. We will have an external call and storage load.

Reply:
No, you're absolutely right, it is not correct if it does not check the `Ash`. Otherwise, someone could `kiss` a little bit of `Ash` before the first `dent` and block `denting`.

Reply:
Ah that's true, you can still `kiss` from generated surplus via the fees or `flip` auctions.

Reply:
Fixed it.